////////////////////////////////////////////////////////////////////////////////
//
//	Includes
//
#include "gui_declinestoreitemdlg.h"

// Qt Includes
#include <QMessageBox>
////////////////////////////////////////////////////////////////////////////////

////////////////////////////////////////////////////////////////////////////////
namespace fm {
////////////////////////////////////////////////////////////////////////////////
namespace gui {
////////////////////////////////////////////////////////////////////////////////


////////////////////////////////////////////////////////////////////////////////
//
// class CAddCustomerDlg
//

// Constructors
CDeclineStoreItem::CDeclineStoreItem( int nAvailableCount, QWidget* pwParent )
	: QDialog( pwParent ),
	  m_nAvailableCount(0)
{
	ui.setupUi( this );	
	setFixedSize(300, 143);
	SetAvailableCount( nAvailableCount );
	ui.sbxTotal->setValue( m_nAvailableCount );
	// Connections
	FM_CONNECT( ui.sbxSubtractCount, valueChanged( int ), this, OnCountChanged( int ) );
	//FM_CONNECT( ui.btnSubtract, clicked(), this, accept() );
	ui.sbxSubtractCount->setValue( 0 );
}

void CDeclineStoreItem::OnCountChanged( int nNewCount )
{
	ui.sbxTotal->setValue( m_nAvailableCount - nNewCount );
	if (nNewCount > 0)
		ui.btnSubtract->setEnabled( true );
	else 
		ui.btnSubtract->setDisabled( true );
}
////////////////////////////////////////////////////////////////////////////////


////////////////////////////////////////////////////////////////////////////////
} // namespace gui
////////////////////////////////////////////////////////////////////////////////
} // namespace fm
////////////////////////////////////////////////////////////////////////////////
